### PR TITLE
Ui cleanup

### PR DESCRIPTION
**Pull request by JakeGinnivan**
This builds on #75, so it should be merged first

### CODE REVIEW

**Comment by JakeGinnivan** [comment]
Will close #68 too hopefully

**Comment by JakeGinnivan** [comment]
Also fixes #62

**Comment by JakeGinnivan** [comment]
image
image

File: src/Carnac.Logic/IConcurrencyService.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+using System.Reactive.Concurrency;
+
+namespace Carnac.Logic
+{
+    public interface IConcurrencyService
+    {
+        IScheduler UiScheduler { get; }
```
Comment (shiftkey) [inline src/Carnac.Logic/IConcurrencyService.cs]: Could we go with `DispatcherScheduler` here (if we're going with idiomatic Rx concepts) or `MainThreadScheduler` here (if we're going with idiomatic ReactiveUI concepts)?

File: src/Carnac.Logic/Models/Processinfo.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+namespace Carnac.Logic.Models
+{
+    public class Processinfo
+    {
+        public Processinfo(string processName)
```
Comment (shiftkey) [inline src/Carnac.Logic/Models/Processinfo.cs]: s/info/Info

File: src/Carnac/CarnacTrayIcon.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                Icon = new Icon(iconStream),
+                ContextMenu = new ContextMenu(new[] { exitMenuItem })
+            };
+
+            exitMenuItem.Click += (sender, args) =>
+            {
+                trayIcon.Visible = false;
+                Application.Current.Shutdown();
+            };
+            trayIcon.DoubleClick += NotifyIconClick;
```
Comment (shiftkey) [inline src/Carnac/CarnacTrayIcon.cs]: I thought I was going crazy testing this, but can we put this back to a single-click?

```
ni.MouseClick += NotifyIconClick;
```

Comment (JakeGinnivan) [inline src/Carnac/CarnacTrayIcon.cs]: yup.

Comment (JakeGinnivan) [inline src/Carnac/CarnacTrayIcon.cs]: Done

File: src/Carnac/Carnac.csproj [hunk 4]
```diff
@@ -58,6 +58,7 @@
     <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
     <CodeAnalysisRuleSetDirectories>;C:\Program Files (x86)\Microsoft Visual Studio 10.0\Team Tools\Static Analysis Tools\\Rule Sets</CodeAnalysisRuleSetDirectories>
     <CodeAnalysisRuleDirectories>;C:\Program Files (x86)\Microsoft Visual Studio 10.0\Team Tools\Static Analysis Tools\FxCop\\Rules</CodeAnalysisRuleDirectories>
+    <UseVSHostingProcess>false</UseVSHostingProcess>
```
Comment (shiftkey) [inline src/Carnac/Carnac.csproj]: :arrow_up:

File: src/Carnac/Carnac.csproj [hunk 5]
```diff
@@ -235,6 +207,7 @@
   <ItemGroup>
     <Resource Include="Resources\Entypo-license.txt" />
   </ItemGroup>
+  <ItemGroup />
```
Comment (shiftkey) [inline src/Carnac/Carnac.csproj]: :fire: damn you csproj files

File: src/Carnac/Carnac.csproj (hunk 4 above)
Comment (JakeGinnivan) [inline src/Carnac/Carnac.csproj]: This one was intentional. VSHosting process put an icon in the tray when there shouldn't be one. Turned off so it was more like when it is being run without VS

File: src/Carnac/UI/PreferencesViewModel.cs [hunk 6]
```diff
@@ -87,16 +107,11 @@ public void Visit()
             }
             catch //I forget what exceptions can be raised if the browser is crashed?
             {
-
+                // ignored
```
Comment (shiftkey) [inline src/Carnac/UI/PreferencesViewModel.cs]: :fire: lol

File: src/Carnac/UI/PreferencesViewModel.cs [hunk 7]
```diff
@@ -87,16 +107,11 @@ public void Visit()
...
             }
         }
 
-        public void SaveSettingsGeneral()
-        {
-            SaveSettings();
-        }
-
-        public void SaveSettings()
+        private void SaveSettings()
```
Comment (shiftkey) [inline src/Carnac/UI/PreferencesViewModel.cs]: :fire: `private`

File: src/Carnac/UI/PreferencesViewModel.cs (hunk 6 above)
Comment (JakeGinnivan) [inline src/Carnac/UI/PreferencesViewModel.cs]: Otherwise we get a warning about empty catch blocks

Comment (JakeGinnivan) [inline src/Carnac/UI/PreferencesViewModel.cs]: Ill move the comment from above down

Comment (shiftkey) [inline src/Carnac/UI/PreferencesViewModel.cs]: Tell the compiler to take a hike

File: src/Carnac/Carnac.csproj (hunk 4 above)
Comment (shiftkey) [inline src/Carnac/Carnac.csproj]: :ok_hand:

File: src/Carnac/UI/PreferencesViewModel.cs [hunk 8]
```diff
@@ -65,7 +82,6 @@ public string Version
                                                        {
                                                          "MahApps.Metro",
                                                          "Fody",
-                                                         "Caliburn Micro",
```
Comment (shiftkey) [inline src/Carnac/UI/PreferencesViewModel.cs]: You should add `Shouldly` in here, given you've mentioned test frameworks below...

**Comment by shiftkey** [comment]
Polish things, :shipit: otherwise

**Pull request merged**